subprojects/packagefiles/libmicrohttpd/meson.build

```diff
@@ -118,7 +118,25 @@ foreach s : ['size_t', 'unsigned int', 'unsigned long long']
   cdata.set('SIZEOF_@0@'.format(s.underscorify().to_upper()), cc.sizeof(s))
 endforeach
 
-cdata.set('SIZEOF_STRUCT_TIMEVAL_TV_SEC', cc.sizeof('time_t', prefix: '#include <time.h>'))
+cdata.set(
```

Comment:
this is garbage and will not work.

Reply:
Well, it works when compiling natively on Windows, Linux and MacOs.
And cross compiling from Linux 64bits to Linux 32 bits, arm (32 and 64 bits), Anroid (various archs).

And it is what is done by upstream libmicrohttpd autotools: https://github.com/Karlson2k/libmicrohttpd/blob/master/configure.ac#L2448-L2474

So can you explain a bit more "garbage" and "will not work" ?

Reply:
> this is garbage and will not work.

Extremly rude and inefficient feedback!

Reply:
You ported an autotools check thinking meson works the same as autotools. Why do you think those macros for those headers are available?

Just switch the check from time_t to long.

Reply:
> Why do you think those macros for those headers are available?

Which macro are you speaking about ?

> Just switch the check from time_t to long.

`long` is not the right type outside of Windows.
My first test was indeed to use `long` when on Windows but it seems proper to me to test the size of `timeval.tv_sec` whatever is its real type to be platform independent.

Reply:
> > Why do you think those macros for those headers are available?
> 
> Which macro are you speaking about ?

All those `HAVE_xxx`: they're not defined, so those `#ifdef HAVE_[…] #endif` blocks are dead code.

Reply:
Ok. New commit (to fixup) simply use `long` on Windows.